engine/src/main/java/org/hibernate/search/engine/mapper/mapping/building/impl/IndexSchemaFilter.java

```diff
@@ -119,28 +123,36 @@ public boolean isPathIncluded(String relativePath) {
 	 */
 	private boolean isPathIncludedInternal(int relativeDepth, String relativePath,
 			boolean markAsEncountered, boolean includedByChild) {
-		boolean includedByThis = depthFilter.isEveryPathIncludedAtDepth( relativeDepth )
-				|| pathFilter.isExplicitlyIncluded( relativePath );
+		boolean includedByThis = ( depthFilter.isEveryPathIncludedAtDepth( relativeDepth )
+				|| pathFilter.isExplicitlyIncluded( relativePath ) )
+				&& !pathFilter.isExplicitlyExcluded( relativePath );
 
 		boolean includedByParent = true;
 		/*
 		 * The parent can filter out paths that are considered as included by a child,
 		 * by reducing the includeDepth in particular,
 		 * but it cannot include paths that are filtered out by a child.
+		 *
+		 * Note: If we only go into recursion when `includedByThis == true` path tracking of parent nodes will not record
+		 * the route.
 		 */
 		if ( parent != null ) {
 			includedByParent = parent.isPathIncludedInternal(
 					relativeDepth + 1,
 					definition.relativePrefix() + relativePath,
-					markAsEncountered, includedByThis
+					markAsEncountered,
+					// need to include a results from the previous step as it may be that we have an include at the current level
+					// but it was excluded at some child level above, and we want to pass it down to the parent.
```

Comment:
"child level above" ? Is that the right term?

Reply:
Not sure I understand what you mean exactly... the "previous child level" is just... the parent?

How about:

```suggestion
					// but it was excluded at some ancestor level above, and we want to pass it up to the parent.
```